src/functional/otf/binding/cpp_interface.py

```diff
@@ -19,6 +19,7 @@
 
 import numpy as np
 
+import functional.otf.binding.type_specifications as ts
```

Comment:
Well, the idea was that the derived modules are a strict superset of the original `type_system.*` modules, and in that case you can just simply import the extended module.
Sounds like it would be kinda nice to use, but to be honest it might be better as you suggest. Re-importing all the stuff is ugly and error-prone, and then not knowing which flavour of the module you are using at the moment is also a problem. Plus, it's really nasty if you want to use two extended modules at the same time.

Anyways, I think this is out of the scope of this PR, but we can change it in a follow-up PR.

Reply:
I agree we should not use the same alias for different modules. I think it would be good enough for this PR to just change the alias to a similar but different one. For example:
```
import functional.otf.binding.type_specifications as bts
```